### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by exectails** [NoCode-NoLife/melia#187] [Discussions]
When one of my server emulators was DMCAd before, we immediately lost access to the entire repo. With Git, we naturally still had all of our code and its history, but we lost all issues and also the entire wiki. Luckily, some of us had a backup, but if that hadn't been the case, we could've lost a lot of data. In addition to a potential data loss if something goes wrong with our repo here, users also always need to go onto GitHub to look up any documentation we put on the wiki. In time we would surely add information about NPCs, scripting, data, and more, and users might want to look things up regularly.

After said DMCA, I switched to putting all documention into markdown files inside the repo's `doc` folder in other projects. This way, you always have all the documentation right there, you can access it anytime you want, even if you don't have internet, you don't necessarily need to go to GitHub, and the documentation can never get lost.

Being able to view the documentation nicely formatted in a browser is great, but I think we might want to also put it into the repo. Or maybe even only into the repo, since you can still navigate to the doc folder on GitHub and open the nicely formatted markdown files there. What do you think?

**Comment by Adephx** [comment]
"When one of my server emulators was DMCAd before, we immediately lost access to the entire repo" Does this also apply if you fork the repo to a private repo, as a measure of backup? For example: https://gist.github.com/0xjac/85097472043b697ab57ba1b1c7530274

**Reply by exectails** [reply]
They will disable the entire repo if they get a DMCA, but it will only affect the repos and forks that were explicitly mentioned in the DMCA. In our case, they didn't DMCA a single one of the forks for some reason, so you could still get the emulator at the time. And since Git is decentralized, we naturally also had the entire repo on our own machines. The GitHub wikis are separate repos though. You can clone them as well, but I don't think that's something a lot of people do. And ours would've been lost if I hadn't cloned it. I don't even remember why I did that^^"

**Comment by ovflowd** [comment]
> With Git, we naturally still had all of our code and its history, but we lost all issues and also the entire wiki. Luckily, some of us had a backup, but if that hadn't been the case, we could've lost a lot of data.

Afaik the repository owner can still access the repository. And extract the data. Also GitHub changed their policies for DMCA. Also, if this happens by any chance, I have contacts at GitHub where we could also appeal.

**Reply by exectails** [reply]
Maybe it has changed in the last 4 years, or there are different severity levels, but in my case I wasn't able to access _anything_ anymore, and GitHub told me it would stay that way unless I were to submit a counter-DMCA.

**Reply by ovflowd** [reply]
Well, afaik that changed... Maybe not in `org-based` repos... Or for older DMCA takedowns...

**Comment by ovflowd** [comment]
Perhaps the Documentation should stay inside the repository as Markdown files as much as possible. We could avoid using the Wiki. But we could also use it. It doesn't matter much 😅

**Reply by exectails** [reply]
Well, the point I was trying to make was more like "if it's in the repo, all potential problems go away completey" xD And you gain the abilitlity to access it offline and whenever, wherever.

**Reply by ovflowd** [reply]
Well for `Issues` and `Discussions` we could move the Issues to another repository altogether with the discussions.

What about creating a repository called `NoCode-NoLife/community` which has 0 code. (Everything from Security, PRs, Actions, Wiki et cetera is disabled) except for Issues and Discussions and "Projects".

And then we could leave all the Issues there ;)

Problem solved. DMCA's cannot happen to repositories without code 😄

**Reply by exectails** [reply]
Hm... that would save the issues, but don't you think it would be nice to have the issue-features available, like linking and auto-closing? And if we were to get DMCAd, and didn't counter it, would the issues still be of any use?

**Reply by ovflowd** [reply]
They would because you could still make just one repo?

AutoClosing and linking works across repositories ;)

**Reply by exectails** [reply]
I'm not sure what you mean. If we're gone, because DMCA, the issues don't have any use for us anymore, do they? Except for if someone wants to continue and would like to have the bug reports.

Didn't know that. How does that work? Surely not with a simple `#123`.
